### ISSUE TITLE
loading “silk-brush” take a long time , several mins

### ISSUE THREAD

**Issue by dingsing2021** [Igalia/wolvic#219]
## Configuration

Wolvic version: 1.0.1
Wolvic build ID: 

Hardware: huawei vr glass 6dof 

## Steps to Reproduce

1. open wolvic 6dof 
2.enter :https://msub2.github.io/silk-brush/
3.

## Current Behavior

take a long time to be loading, several mins

## Expected Behavior

loading time should be seconds

## Error Logs and Stack Traces

<details open>

```
```

</details>

**Comment by svillar** [comment]
Seems an specific HVR issue as it does not happen on Quest2

**Comment by msub2** [comment]
I'm the developer that made Silk Brush. The initial download of Silk Brush is slightly over 100 MB, most of it being the WASM blob from Unity, so that might also have something to do with it.

**Comment by svillar** [comment]
> I'm the developer that made Silk Brush. The initial download of Silk Brush is slightly over 100 MB, most of it being the WASM blob from Unity, so that might also have something to do with it.

Yeah, I noticed it download a lot of data. Quest2 is capable of dealing with that, but perhaps that is not the case with a Huawei phone. Is there any possibility to somehow split the resources? You mentioned that most of it is the wasm blob from Unity so I guess it isn't... ?

**Comment by msub2** [comment]
> > I'm the developer that made Silk Brush. The initial download of Silk Brush is slightly over 100 MB, most of it being the WASM blob from Unity, so that might also have something to do with it.
> 
> Yeah, I noticed it download a lot of data. Quest2 is capable of dealing with that, but perhaps that is not the case with a Huawei phone. Is there any possibility to somehow split the resources? You mentioned that most of it is the wasm blob from Unity so I guess it isn't... ?

I'm honestly not entirely sure at the moment, it became a too much hassle to keep developing (lots of stuff from the original Tilt Brush source code doesn't play as nicely with WebGL and compilation times are unreasonably high) so I haven't come back to the project in some time. It's possible I could separate out some of the resources into assetbundles, but I'd need to spend a day with it to see how much that would actually help.
